clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonLoadBalancerProvider.groovy

```diff
@@ -74,7 +74,15 @@ class AmazonLoadBalancerProvider implements LoadBalancerProvider<AmazonLoadBalan
 
     CacheData application = cacheView.get(APPLICATIONS.ns, Keys.getApplicationKey(applicationName))
 
-    Collection<CacheData> applicationServerGroups = application ? resolveRelationshipData(application, SERVER_GROUPS.ns) : []
+    Collection<CacheData> applicationServerGroups = []
+    if (application) {
+      applicationServerGroups = resolveRelationshipData(
+        application,
+        SERVER_GROUPS.ns,
+        RelationshipCacheFilter.include(INSTANCES.ns, LOAD_BALANCERS.ns, TARGET_GROUPS.ns)
+      )
+    }
+
     Collection<String> allLoadBalancerKeys = cacheView.getIdentifiers(LOAD_BALANCERS.ns)
     Collection<String> allTargetGroupKeys = cacheView.getIdentifiers(TARGET_GROUPS.ns)
```

Comment:
I'd like to see the need for allLoadBalancerKeys and allTargetGroupKeys removed too but this is a great start.